lib/memo_wise.rb

```diff
@@ -36,6 +37,7 @@ def memo_wise(method_name)
           def #{method_name}(*args)
             key = [:#{method_name}, args].freeze
             @_memo_wise.fetch(key) do
+              @_memo_wise_keys[:#{method_name}] << args
```

Comment:
I'm curious (not saying this isn't the best approach, I really don't know) to learn more about why you went down this route. I can see a few different implementation paths:

1. What you have here.
2. Delete this line. On reset, iterate through all keys and delete the ones that start with `:#{method_name}`.
3. Same as 2, but change keys to be another level of a Hash rather than an array. On reset, just delete the key.

Should we proceed with this as-is and revisit it when we have benchmarks merged? Should we add benchmarks for `reset_memo_wise` at all?

Reply:
I was thinking about different paths here and totally agree with your thought about re-visiting when benchmarks are merging. Since this gem is attempting to optimize for speed, that seems like the best criteria with which to make this decision!

I can look too, but out of curiosity, do you remember most other gems having reset methods too?

Reply:
Yep I think we'd probably want benchmarks for these worst-case scenarios:

- the case where memoized methods are getting called with lots of different arguments
- the case where method results are getting memoized and reset repeatedly